### PR TITLE
fix timezone issue with version timestamp

### PR DESCRIPTION
**Pull request by changhiskhan**
make the user be explicit about timestamp

### CODE REVIEW

File: python/lance/tests/test_schema_evolution.py [hunk 1]
```diff
@@ -68,13 +69,11 @@ def test_add_column_with_literal(tmp_path: Path):
...
-        "timestamp"
-    ] > datetime.datetime.utcnow() - datetime.timedelta(0, 10)
-    # Let Python auto convert timezones
-    assert dataset.version[
-               "timestamp"
-           ] > datetime.datetime.now() - datetime.timedelta(0, 10)
+    ts = dataset.version["timestamp"]
+    with pytest.raises(TypeError):
+        # can't compare offset-naive and offset-aware datetimes
+        assert ts > datetime.datetime.utcnow() - datetime.timedelta(0, 10)
```
Comment (eddyxu) [inline python/lance/tests/test_schema_evolution.py:75]: if the first assert raise exception, the second assert won't run?

File: python/lance/tests/test_schema_evolution.py [hunk 2]
```diff
@@ -68,13 +69,11 @@ def test_add_column_with_literal(tmp_path: Path):
...
     # Created within 10 seconds, in UDT
-    assert dataset.version[
-        "timestamp"
-    ] > datetime.datetime.utcnow() - datetime.timedelta(0, 10)
-    # Let Python auto convert timezones
-    assert dataset.version[
-               "timestamp"
-           ] > datetime.datetime.now() - datetime.timedelta(0, 10)
+    ts = dataset.version["timestamp"]
+    with pytest.raises(TypeError):
```
Comment (eddyxu) [inline python/lance/tests/test_schema_evolution.py:73]: Can we keep a test to make sure `timestamp` is not `unset / 0` , like `1970-XX-XX`?

**Review by eddyxu — COMMENTED** [review]
Minor issues

File: python/lance/tests/test_schema_evolution.py (hunk 1 above)
Comment (eddyxu) [reply python/lance/tests/test_schema_evolution.py:75]: nvm. i thought the line 76 is also protected by `pytest.raise` .

File: python/lance/tests/test_schema_evolution.py (hunk 2 above)
Comment (changhiskhan) [reply python/lance/tests/test_schema_evolution.py:73]: added

File: python/lance/tests/test_schema_evolution.py (hunk 1 above)
Comment (changhiskhan) [reply python/lance/tests/test_schema_evolution.py:75]: discussed. added newline for readability

**Review by eddyxu — APPROVED** [review]
Thanks! LGTM

**Pull request merged**